okio/src/jvmMain/kotlin/okio/aio/AsyncGzipSource.kt

```diff
@@ -0,0 +1,180 @@
+package okio.aio
+
+import java.io.EOFException
+import java.io.IOException
+import java.util.zip.*
+import okio.*
+
+class AsyncGzipSource(source: RawSource) : RawSource {
```

Comment:
FYI: it is a copy-pasted version of `GzipSource` where I changed `read` method and `awaitContent`.

Making Gzip full async is much easier than supporting a mixed mode :)